fawltydeps/extract_declared_dependencies.py

```diff
@@ -288,6 +288,28 @@ def parse_pyproject_toml(path: Path) -> Iterator[DeclaredDependency]:
     else:
         logger.debug("%s does not contain [tool.poetry].", source)
 
+    if "dynamic" in parsed_contents.get("tool", {}).get("setuptools", {}).keys():
+        dynamic_section = parsed_contents["tool"]["setuptools"]["dynamic"]
+        dynamic_deps_files = dynamic_section.get("dependencies", {}).get("file", [])
+        dynamic_optional_deps_files = [
+            file_list
+            for optional_deps_item in [
+                optional_deps_section.get("file", {})
+                for optional_deps_section in dynamic_section.get(
+                    "optional-dependencies", {}
+                ).values()
+            ]
+            for file_list in optional_deps_item
+        ]
+        dynamic_files = dynamic_deps_files + dynamic_optional_deps_files
+        if isinstance(dynamic_files, list):
```

Comment:
Why is this test necessary?

Reply:
Closing this as it's addressed in [72aa2cd](https://github.com/tweag/FawltyDeps/pull/351/commits/72aa2cd1834e29b2bf9e0237cf2c9489b28c999b).